### PR TITLE
Work towards M1 support

### PR DESCRIPTION
**Pull request by chrisseaton**
This leaves us blocked by Sulong, with

```
<no message> (com.oracle.truffle.api.CompilerDirectives.ShouldNotReachHere)
        from com.oracle.truffle.api.CompilerDirectives.shouldNotReachHere(CompilerDirectives.java:570)
        from com.oracle.truffle.api.CompilerDirectives.shouldNotReachHere(CompilerDirectives.java:490)
        from com.oracle.truffle.llvm.parser.factories.UnknownBasicPlatformCapability.getVAListType(UnknownBasicPlatformCapability.java:78)
        from com.oracle.truffle.llvm.parser.factories.BasicNodeFactory.<init>(BasicNodeFactory.java:382)
        from com.oracle.truffle.llvm.nativemode.NativeConfiguration.createNodeFactory(NativeConfiguration.java:69)
        from com.oracle.truffle.llvm.runtime.pthread.LLVMPThreadContext.lambda$new$0(LLVMPThreadContext.java:92)
        from com.oracle.truffle.llvm.runtime.LLVMLanguage.lambda$createCachedCallTarget$0(LLVMLanguage.java:633)
        from java.base/java.util.concurrent.ConcurrentHashMap.computeIfAbsent(ConcurrentHashMap.java:1705)
        from com.oracle.truffle.llvm.runtime.LLVMLanguage.createCachedCallTarget(LLVMLanguage.java:633)
        from com.oracle.truffle.llvm.runtime.pthread.LLVMPThreadContext.<init>(LLVMPThreadContext.java:91)
        from com.oracle.truffle.llvm.runtime.LLVMContext.<init>(LLVMContext.java:233)
        from com.oracle.truffle.llvm.runtime.LLVMLanguage.createContext(LLVMLanguage.java:312)
        from com.oracle.truffle.llvm.runtime.LLVMLanguage.createContext(LLVMLanguage.java:84)
        from com.oracle.truffle.api.LanguageAccessor$LanguageImpl.createEnvContext(LanguageAccessor.java:263)
        from com.oracle.truffle.polyglot.PolyglotLanguageContext.ensureCreated(PolyglotLanguageContext.java:587)
        from com.oracle.truffle.polyglot.EngineAccessor$EngineImpl.lookupService(EngineAccessor.java:1106)
        from com.oracle.truffle.api.TruffleLanguage$Env.lookup(TruffleLanguage.java:2412)
        from org.truffleruby.language.TruffleBootNodes.getToolchain(TruffleBootNodes.java:444)
        from org.truffleruby.language.TruffleBootNodes$ToolchainExecutableNode.toolchainExecutable(TruffleBootNodes.java:397)
        from org.truffleruby.language.TruffleBootNodesFactory$ToolchainExecutableNodeFactory$ToolchainExecutableNodeGen.executeAndSpecialize(TruffleBootNodesFactory.java:981)
        from org.truffleruby.language.TruffleBootNodesFactory$ToolchainExecutableNodeFactory$ToolchainExecutableNodeGen.execute(TruffleBootNodesFactory.java:966)
        from org.truffleruby.language.RubyCoreMethodRootNode.execute(RubyCoreMethodRootNode.java:48)
        from com.oracle.truffle.api.impl.DefaultCallTarget.callDirectOrIndirect(DefaultCallTarget.java:85)
/Users/chrisseaton/Documents/graalvm/truffleruby/lib/truffle/rbconfig.rb:65:in `toolchain_executable'
        from /Users/chrisseaton/Documents/graalvm/truffleruby/lib/truffle/rbconfig.rb:65:in `<module:RbConfig>'
        from /Users/chrisseaton/Documents/graalvm/truffleruby/lib/truffle/rbconfig.rb:37:in `<top (required)>'
        from <internal:core> core/kernel.rb:260:in `require'
        from extconf.rb:16:in `<main>'
```

The naming seems inconsistent - sometimes `aarch64` sometimes `arm64`, but I'm using the names that the tools generate for me!

### CODE REVIEW

**Comment by lewurm** [comment]
FYI I'm working on this (internal issue GR-37257). I came up with a similar patch: https://github.com/lewurm/truffleruby/commits/burbanfo/darwin-aarch64-bringup

I'm also working on the mentioned Sulong issue (GR-36936), this is my "it's building, but completely wrong" branch: https://github.com/lewurm/graal-1/tree/burbanfo/darwin-aarch64-sulong-platform
As soon as Sulong hits some va_args like function it will either return random values or crash.

File: src/main/c/truffleposix/truffleposix.c [hunk 1]
```diff
@@ -467,7 +467,7 @@ static void copy_stat(struct stat *native_stat, struct truffleposix_stat* buffer
 #elif defined(HAVE_STRUCT_STAT_ST_ATIMENSEC)
   buffer->atime_nsec = (long)native_stat->st_atimensec;
 #else
-  buffer->atime_nsec = 0
+  buffer->atime_nsec = 0;
```
Comment (eregon) [inline src/main/c/truffleposix/truffleposix.c:470]: @lewurm This is the difference with your branch, would be good to pick it there

**Comment by eregon** [comment]
Thank you, but I think we should only add aarch64 stuff once we can test it in CI, and that means we need Sulong to get aarch64 support first (we did the opposite for linux-aarch64, and it was significant overhead). So I'll use @lewurm's branch because that's easier to adjust, update and test in CI, once dependencies are all aarch64-ready.

**Comment by eregon** [comment]
> The naming seems inconsistent - sometimes `aarch64` sometimes `arm64`, but I'm using the names that the tools generate for me!

Yes, I think we should consolidate to aarch64 as a cleanup, that seems the standardized name for it (even though it does look/sound weird for people not knowing aarch64 yet, but one advantage is it's visually more distinct to amd64).

File: src/main/c/truffleposix/truffleposix.c (hunk 1 above)
Comment (chrisseaton) [reply src/main/c/truffleposix/truffleposix.c:470]: I saw the syntax error from this in practice on my machine.

**Pull request closed without merging**